tests/api2/test_530_ups.py

```diff
@@ -61,38 +136,31 @@ def test_04_Set_UPS_options():
         'driver': 'usbhid-ups$PROTECT NAS',
         'monpwd': 'mypassword'
     }
-    results = PUT('/ups/', payload)
-    assert results.status_code == 200, results.text
+    results = call('ups.update', payload)
 
 
 @pytest.mark.parametrize('data', first_ups_list)
 def test_05_look_at_UPS_options_output_of_(data):
-    assert payload[data] == results.json()[data], results.text
+    assert payload[data] == results[data], results.text
 
 
 def test_06_starting_ups_service():
-    payload = {
-        "service": "ups",
-    }
-    results = POST('/service/start/', payload)
-    assert results.status_code == 200, results.text
+    call('service.start', 'ups')
 
 
 def test_07_look_UPS_service_status_is_running():
-    results = GET(f'/service/id/{ups_id}/')
-    assert results.status_code == 200, results.text
-    assert results.json()['state'] == 'RUNNING', results.text
+    results = call('service.query', [['service', '=', 'ups']], {'get': True})
+    assert results['state'] == 'RUNNING', results
 
 
 def test_08_get_API_reports_UPS_configuration_as_saved():
```

Comment:
Another example of a function that uses `global` keyword for no apparent reason. Furthermore, the next few tests that test the `results` object can be condensed into 1 function. We're making multiple `ups.update` calls (which is fine) but we can test all the changes in 1 function.